### ISSUE TITLE
Dart SDK version (2.1.0) is not supported by geolocation

### ISSUE THREAD

**Issue by isoos** [loup-v/geolocation#44]
_From @markflarup on September 5, 2018 8:35_

URL: https://pub.dartlang.org/packages/geolocation#-installing-tab-

I want to support geo location in an app, but after adding `geolocation: ^0.2.1` as a dependency the following error message is returned:

```
The current Dart SDK version is 2.1.0-dev.0.0.flutter-be6309690f.

Because [APP] depends on geolocation >=0.1.1 which requires SDK version <2.0.0, version solving failed.
pub get failed (1)
```

Do you have a estimate for when it will be supported?

_Copied from original issue: dart-lang/pub-dartlang-dart#1580_

**Comment by isoos** [comment]
@markflarup this repository is for pub site, please file issues to the package's repository or contact the developer directly. I'm moving the issue.

**Comment by velegip** [comment]
I think you can use this branch : https://github.com/driveddy/geolocation

**Comment by markflarup** [comment]
> I think you can use this branch : https://github.com/driveddy/geolocation

I will look into it, thanks! :)
As of now I have relied on geolocator (https://pub.dartlang.org/packages/geolocator)

**Comment by mi6crazyheart** [comment]
My app also broke due to this Dart SDK version incompatibility. Guys, any time frame by when it'll get the fix? -.-

**Comment by jiachen247** [comment]
upvote! This sdk change is breaking everything.

**Comment by jiachen247** [comment]
geolocation:
    git: "https://github.com/loup-v/geolocation.git"

this worked for me. i think the dart package system is doing additional checks for dart 2 compatibility.

**Comment by shehabic** [comment]
I created a fork of this, that fixes android as well as adding an option to enable Location Services in case it's disabled on Android.
https://github.com/shehabic/geolocation
you can use it as
```
geolocation:
   git:
      url: 'https://github.com/shehabic/geolocation'
```

**Comment by vashisth00** [comment]
Where do we add this? in pubspec.yaml? as a new dependency?

**Comment by lukaspili** [comment]
Sorry for the delay. It should be fixed in the latest release 1.0.0.
